From 46c7fe11bd0ed8595c3f920d42a9914fa864d893 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sun, 7 Feb 2021 18:51:14 +0100
Subject: [PATCH] gdk-pixbuf/gdk-pixbuf-io.c: fix build without gmodule

Fix the following build failure:

../gdk-pixbuf/gdk-pixbuf-io.c: In function 'gdk_pixbuf_io_init':
../gdk-pixbuf/gdk-pixbuf-io.c:681:16: error: implicit declaration of function 'gdk_pixbuf_get_module_file'; did you mean '_gdk_pixbuf_get_module'? [-Werror=implicit-function-declaration]
  681 |  module_file = gdk_pixbuf_get_module_file ();
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                _gdk_pixbuf_get_module

Fixes:
 - http://autobuild.buildroot.org/results/6cd54c497f5d19342ec94ece713547b887e4c02d

Upstream status: Accepted
https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/merge_requests/103

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 gdk-pixbuf/gdk-pixbuf-io.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gdk-pixbuf/gdk-pixbuf-io.c b/gdk-pixbuf/gdk-pixbuf-io.c
index 40eb920c8..2dc2ea6da 100644
--- a/gdk-pixbuf/gdk-pixbuf-io.c
+++ b/gdk-pixbuf/gdk-pixbuf-io.c
@@ -678,7 +678,9 @@ gdk_pixbuf_io_init (void)
 	gboolean ret;
 
 	gdk_pixbuf_io_init_builtin ();
+#ifdef USE_GMODULE
 	module_file = gdk_pixbuf_get_module_file ();
+#endif
 	ret = gdk_pixbuf_io_init_modules (module_file, NULL);
 	g_free (module_file);
 	return ret;
-- 
2.29.2

